### PR TITLE
JP-3354 Cube build emsm bug

### LINKED ISSUE #7841: Cube build using emsm fails on some MAC  computers
_Issue [JP-3354](https://jira.stsci.edu/browse/JP-3354) was created on JIRA by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison):_

Cube_build fails to create a ifu cube when weighting='emsm' for  MAC computers. On the systems tested the failure occurs on  M1 and M2 computers. The error produces a segmentation fault. The error has been tracked to the c extension code:  cube_match_sky_pointcloud.c. When the dq_miri and dq_nirspec code is called from the program a variable is mistakenly passed as an int instead of a long.

### PR DESCRIPTION
**Pull request by jemorrison**
Resolves [JP-3354](https://jira.stsci.edu/browse/JP-3354)

Closes #7841 

This PR fixes a bug in cube_build when creating an IFU cube using weighting='emsm'.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)

https://stsci.service-now.com/nav_to.do?uri=%2Fincident.do%3Fsys_id%3Dced5d8024778b510ec5b9448436d43ac%26sysparm_stack%3Dincident_list.do%3Fsysparm_query%3Dactive%3Dtrue%26sysparm_view%3DJWST%26sysparm_view_forced%3Dtrue
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by jemorrison** [comment]
@Howard I ran the regression tests and the new tests just should differences in the data I put on artifcatory that I created on my MAC with those produced by the regression test. They are small differences so I think the PR is ready.
Not sure about the check-style tests - that was not from this PR but I will rebase and see if that cleans it up

File: CHANGES.rst [hunk 1]
```diff
@@ -44,6 +44,8 @@ cube_build
 - Correct slicer scale and orientation in output WCS for IFU cubes built in internal_cal
   coordinates, for NIRSpec calibration analysis. [#7811]
 
+- Fix a bug when weighting=emsm. [#7847]
```
Comment (hbushouse) [inline CHANGES.rst:47]:
```suggestion
- Fix a bug with memory allocation in C extensions when weighting=emsm. [#7847]
```

**Review by hbushouse — COMMENTED** [review]
Code updates look fine. Have the new regtest data files been uploaded to Artifactory?

**Review by hbushouse — APPROVED** [review]
I found the regtest run. Looks good.

**Pull request merged**